#!/usr/bin/env python
PACKAGE = "mav_linear_mpc"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("roll_gain",    double_t,    0, "proportional gain of roll angle",   90.0 ,  0.1, 200.0)
gen.add("pitch_gain",   double_t,    0, "proportional gain of pitch angle",  80.0 ,  0.1, 200.0)
gen.add("roll_int_gain",   double_t,    0, "integral gain of roll angle",  1.0 ,  0.1, 200.0)
gen.add("pitch_int_gain",   double_t,   0, "integral gain of pitch angle",  1.0 ,  0.1, 200.0)
gen.add("p_gain",       double_t,    0, "derivative gain of roll angle",     30.0 ,  0.1, 200.0)
gen.add("q_gain",       double_t,    0, "derivative gain of pitch angle",    25.0 ,  0.1, 200.0)
gen.add("r_gain",       double_t,    0, "derivative gain of yaw angle",      0.80 ,  0.1, 200.0)

exit(gen.generate(PACKAGE, "Config", "PIDAttitude"))
